src/main/java/com/customerApi/api/services/CustomerService.java

```diff
@@ -114,37 +140,53 @@ public void deleteCustomer(Customer record) {
 			if (!customer.isPresent()) {
 
 				logger.info("Enter valid cutomer record");
-				throw new CustomerNotFoundException("No customer record found");
+				throw new CustomerNotFoundException("Customers record not found");
 
 			}
 			customerRepo.delete(record);
 
 			logger.trace("All Customers have been deleted successfully");
-		} catch (JDBCConnectionException e) {
+		} catch (CannotCreateTransactionException e) {
 			logger.error("Database connection error" + e);
-
-		} catch (Exception e) {
+			throw new CustomerNotFoundException("Database Connectivity Error", e);
+		} 
+		
+		catch (EmptyResultDataAccessException e) {
+			logger.error("Customers record is empty" + e);
+			throw new CustomerNotFoundException("Customer record is empty", e);
+		} 
+		
+		catch (Exception e) {
 			logger.error("Customers have not been deleted " + e);
+			throw new CustomerNotFoundException("Exception in customers deletion", e);
 		}
 
 	}
 
 	public Customer updateCustomer(Customer data) {
-		Optional<Customer> customer = Optional.ofNullable(data);
+		Customer customer = new Customer();
 		try {
-			if (!customer.isPresent()) {
-				throw new CustomerNotFoundException("No customer record found");
+			if (!Optional.ofNullable(data).isPresent()) {
+				throw new CustomerNotFoundException("Customer record not found");
```

Comment:
Not sure why you keep adding Optional everywhere if you already doing validation on controller on customer object,